apps/mobile/src/slices/auth/actions.ts

```diff
@@ -11,11 +11,16 @@ import {
 import { name as sliceName } from "./auth.slice";
 
 const getAuthenticatedUser = createAsyncThunk<
-	UserDto,
+	null | UserDto,
 	undefined,
 	AsyncThunkConfig
 >(`${sliceName}/get-authenticated-user`, async (_, { extra }) => {
 	const { authApi } = extra;
+	const hasToken = await storage.has(StorageKey.TOKEN);
+
+	if (!hasToken) {
+		return null;
+	}
```

Comment:
I think it's better to add `getAuthenticatedUser.rejected` to auth.slice

Reply:
I agree that in addition to `getAuthenticatedUser.rejected,` it would also be helpful to add `getAuthenticatedUser.pending` to the auth.slice. However, there's one point to consider: this doesn't relate to issue #219 fix(mobile): an error message appears after tapping on the 'create an account' link. The implementation I've proposed for issue #219 addresses the problem of sending unnecessary requests to the server. Sure, I can add the `getAuthenticatedUser.pending` and `getAuthenticatedUser.rejected` states to the `auth.slice` if you can explain how they relate to issue #219. 😉

Reply:
I was just thinking of a way to avoid getting the token inside the action. You are right, we can just omit sending the request.